Wix case study (#4841)
GoogleChrome/web.dev: src/site/content/en/blog/wix/index.md

---
layout: post
title: How Wix improved website performance by evolving their infrastructure
subhead: >
  A case study of some major changes introduced at Wix to improve website loading

>   performance for millions of sites, clearing the path for them to receive good PSI and

Author: ```suggestion
  performance for millions of sites, clearing the path for them to receive good PageSpeed Insights and
```

  Core Web Vitals scores.
authors:
  - alonko
description: >
  A case study of some major changes introduced at Wix to improve website loading
  performance for millions of sites, clearing the path for them to receive good PageSpeed Insights and
  Core Web Vitals scores.
date: 2021-03-09
hero: image/BrQidfK9jaQyIHwdw91aVpkPiib2/HNGPDotyTYOuPE0YxLQ9.jpg
alt: "A fast train (source: https://unsplash.com/photos/60VrGk-bfeA)"
tags:
  - blog
  - fast
  - performance
  - web-vitals
  - case-study
---

{% Aside %}

>   Alon leads the backend team at Wix.

Reviewer: ```suggestion
  Alon leads the core backend team at [Wix](https://www.wix.com).
```

     <p class="w-stat__desc">Reduction of median file transfer size</p>
    </div>
  </div>
</div>

Traditionally, all our files were compressed using [gzip
compression](https://en.wikipedia.org/wiki/Gzip), which is the most prevalent
HTML compression option on the web. This compression protocol was initially
implemented almost 30 years ago!

<figure class="w-figure w-figure--inline-right">
  {% Img src="image/BrQidfK9jaQyIHwdw91aVpkPiib2/h7KzeAfg2THMdDGMYvlR.jpg", alt="Brotli compression", width="800", height="805" %}
  <figcaption class="w-figcaption">
    <a href="https://tools.paulcalvano.com/compression.php">Brotli Compression Level Estimator</a>
  </figcaption>
</figure>

The newer [Brotli compression](https://en.wikipedia.org/wiki/Brotli) introduces
compression improvements with almost no trade-offs, and is slowly becoming more
popular, as described in the yearly Web Almanac [Compression
chapter](https://almanac.httparchive.org/en/2020/compression#what-type-of-content-should-we-compress).
It has been supported by [all the major browsers](https://caniuse.com/brotli)
for a while.

We enabled Brotli support on our nginx proxies in the edges, for all clients
that support it.

> Moving to use Brotli compression reduced our median file transfer sizes by **21%

Reviewer: Can't break line here - it breaks the bold of 21%-25% (see in deploy preview)

Author: Sharp eye, thanks for catching

> Moving to use Brotli compression reduced our median file transfer sizes by **21%
> - 25%** resulting in a reduced bandwidth usage and improved loading times.

Author: ```suggestion
Moving to use Brotli compression reduced our median file transfer sizes by **21% to 25%**
resulting in a reduced bandwidth usage and improved loading times.
```

monitoring
[CrUX](https://developers.google.com/web/tools/chrome-user-experience-report/)
reports for our sites as well as our internal performance metrics.

We are committed to continuously improve loading times and provide our users
with a platform where they can build sites as they imagine, without compromising
on performance.

<figure class="w-figure">
  {% Img src="image/BrQidfK9jaQyIHwdw91aVpkPiib2/AADz7d1yVOWZlq0iSb6P.jpg", alt="LCP, Speed Index and FCP for a mobile site over time", width="800", height="259" %}
  <figcaption class="w-figcaption">LCP, Speed Index and FCP for a mobile site over time</figcaption>
</figure>

DebugBear recently released a very interesting [Website Builder Performance
Review](https://www.debugbear.com/blog/website-builder-performance-review),
which touches on some of the areas I mentioned above and examines the
performance of very simple sites built on each platform. This
[site](https://matt05041.wixsite.com/bizsolutions) was built almost two years
ago, and not modified since, but the platform is continually improving, and the
site performance along with it, which can be witnessed by [viewing its

> data](https://www.debugbear.com/project/175/pageLoad/873/overview?dateRange=2019-03-31T21%3A00Z-to-2021-02-10T21%3A59Z)

Reviewer: ```suggestion
data](https://www.debugbear.com/project/175/pageLoad/873/overview?dateRange=2019-03-31T21%3A00Z-to-2021-03-31T21%3A59Z)
```
